src/modules/esl-event-listener/README.md

```diff
@@ -1,50 +1,211 @@
+
+In addition to that ESL allows you to combine the  `ESLEventDesriptor` data with the handler function.
+`ESLEventDesriptorFn` is a function handler that is decorated with the `ESLEventDesriptor` properties.
+
+Here is the list of supported keys of `ESLEventDesriptor`:
+
+- #### `event` key
+  <u>Type:</u> `string | PropertyProvider<string>`  
+  <u>Description:</u> the event type for subscription. 
+  Can be provided as a string or via provider function that will be called in the runtime before exact subscription.
+
+  The event string (as literal or returned by `PropertyProvider`) can declare multiple event types separated by space symbol.
+  ESL will create multiple subscriptions `ESLEventListener` object for each event separately in this case.  
+
+- #### `target` key
+  <u>Type:</u> `string | EventTarget | EventTarget[] | PropertyProvider<string | EventTarget | EventTarget[]>`  
+  <u>Default Value:</u> `host` object itself or `$host` key of the `host` object  
+  <u>Description:</u> the key to declare exact EventTarget for subscription.
+  In case the `target` key is a string then it considered as a [`TraversingQuery`](../esl-traversing-query/README.md) to find 
+  a target relatively to `host | host.$host` object or absolutely in bounds of the active DOM tree.
+  The `target` key is also supports an exact reference(s) for `EventTargets`. 
+
+  ⚠ Any `EventTarget` or event ESL `SynteticEventTarget` (including [`ESLMediaQuery`](../esl-media-query/README.md)) 
+  can be a target for listener API.
+  
+  The `target` property can be declared via `PropertyProvider` as well.  
+  
+- #### `selector` key
+  <u>Type:</u> `string | PropertyProvider<string>`  
+  <u>Default Value:</u> `null`  
+  <u>Description:</u> the CSS selector to check the event bubbling chain.
+  The property to use OOTB event delegation check of target.
+  
+  Supports `PropertyProvider` to declare the computed value as well.
+  
+- #### `capture` key
+  <u>Type:</u> `boolean`  
+  <u>Default Value:</u> `false`  
+  <u>Description:</u> marker to use capturing phase of the DOM event to handle.
+ 
+- #### `passive` key
+  <u>Type:</u> `boolean`  
+  <u>Default Value:</u> `false`  
+  <u>Description:</u> marker to use passive subscription to the native event.
+
+  ⚠ ESL uses passive subscription by default for `wheel`, `mousewheel`, `touchstart`, `touchmove` events.
+  You need to declare `passive` key explicitly to override such behavior.
+  
+- #### `once` key
+  <u>Type:</u> `boolean`  
+  <u>Default Value:</u> `false`  
+  <u>Description:</u> marker to unsubscribe listener with the first successful handling of the event.
+
+- #### `auto` key (for `ESLEventDesriptorFn` declaration only)
+  <u>Type:</u> `boolean`  
+  <u>Default Value:</u> `false` for `ESLEventUtils.initDescriptor`, `true` for `@listen`  
+  <u>Description:</u> marker to make an auto-subscribable descriptor. See [Automatic (collectible) descriptors](#automatic-collectible-descriptors).
```

Comment:
```suggestion
  <u>Description:</u> marker to make an auto-subscribable descriptor. See [Automatic (collectable) descriptors](#automatic-collectable-descriptors).
```
The `collectible` is a noun and the `collectable` is an adjective, isn't it?
Not sure about `subscribable ` =)